File: CS/Entities/Discount.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DataStorage;

namespace Entities
{
    public class Discount : Entity
    {
        private int clientPointsRequired;
        private string description;
        private int productId;
        private double factor;

        public Discount(int points, string description, int productId, double factor)
        {
            this.clientPointsRequired = points;
            this.description = description;
            this.productId = productId;
            this.factor = factor;
        }

        public int ClientPointsRequired { get => clientPointsRequired; set => clientPointsRequired = value; }
        public string Description { get => description; set => description = value; }
        public int ProductId { get => productId; set => productId = value; }
        public double Factor { get => factor; set => factor = value; }

        public void addToBD() {
            List<Object> listParameters = new List<Object>();
            listParameters.Add((Object)clientPointsRequired);
            listParameters.Add((Object)description);
            listParameters.Add((Object)factor);
            listParameters.Add((Object)productId);

            BDManager bdm = BDManager.getInstance();
            bdm.AddUpdate("addToBDDiscount", listParameters);
        }

        public void deleteInBD() {

        }

        public void updateInBD() {

        }
    }
}

File: SistemaJava/src/Entities/Discount.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entities;

/**
 *
 * @author alulab14
 */
public class Discount implements Entity {
    
    private int id;
    private int clientPointsRequired;
    private String description;
    private int productId;
    private double factor;
    private int state;

    public Discount(int id, int clientPointsRequired, String description, int productId, double factor, int state) {
        this.id = id;
        this.clientPointsRequired = clientPointsRequired;
        this.description = description;
        this.productId = productId;
        this.factor = factor;
        this.state = state;
    }

    /**
     * @return the clientPointsRequired
     */
    public int getClientPointsRequired() {
        return clientPointsRequired;
    }

    /**
     * @param clientPointsRequired the clientPointsRequired to set
     */
    public void setClientPointsRequired(int clientPointsRequired) {
        this.clientPointsRequired = clientPointsRequired;
    }

    /**
     * @return the description
     */
    public String getDescription() {
        return description;
    }

    /**
     * @param description the description to set
     */
    public void setDescription(String description) {
        this.description = description;
    }

    /**
     * @return the productId
     */
    public int getProductId() {
        return productId;
    }

    /**
     * @param productId the productId to set
     */
    public void setProductId(int productId) {
        this.productId = productId;
    }
    
    @Override
    public void addToBD() {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }
    
    @Override
    public void updateInBD(){        
        System.out.println("updateInBD ejecutado");
    }
    
    @Override
    public void deleteInBD(){
        System.out.println("deleteInBD ejecutado");        
    }

    /**
     * @return the id
     */
    public int getId() {
        return id;
    }

    /**
     * @param id the id to set
     */
    public void setId(int id) {
        this.id = id;
    }

    /**
     * @return the factor
     */
    public double getFactor() {
        return factor;
    }

    /**
     * @param factor the factor to set
     */
    public void setFactor(double factor) {
        this.factor = factor;
    }

    /**
     * @return the state
     */
    public int getState() {
        return state;
    }

    /**
     * @param state the state to set
     */
    public void setState(int state) {
        this.state = state;
    }

    
    
}
